scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,272 @@
+tar -xf $NAME
+# This gets unpacked into `raw_feature_bc_matrix`
+
+
+# For posterity, first attempt was with MALT, but it had wildly low ADT counts
+#NAME=malt_10k_protein_v3_raw_feature_bc_matrix.tar.gz
+#wget https://cf.10xgenomics.com/samples/cell-exp/3.0.0/malt_10k_protein_v3/$NAME
+```
+
+## Filtering and normalization
+
+```{r}
+
+pbmc_10x_dir <- "raw_feature_bc_matrix"
+# for now..
+mito_file <- "https://raw.githubusercontent.com/AlexsLemonade/scpca-downstream-analyses/main/reference-files/Homo_sapiens.GRCh38.104.mitogenes.txt"
+
+
+
+pbmc_sce_raw <- DropletUtils::read10xCounts(pbmc_10x_dir)
+mito_genes <- readr::read_tsv(mito_file, 
+                              col_names=FALSE) %>%
+  # only keep relevant mitochondrial genes
+  dplyr::filter(X1 %in% rownames(pbmc_sce_raw)) %>%
+  dplyr::pull(X1)
+```
+
+
+### Initial Mitochondria QC:
+
+Note that the `addPer<X>QC()` functions should be run before we split out the ADTs into their own altExp so that the appropriate QC columns actually get ported into the altExp.
+
+```{r}
+pbmc_sce_percellqc <- scater::addPerCellQC(
+  pbmc_sce_raw, 
+  subsets = list(mito = mito_genes))
+
+# we'll need this later, so pop it in now
+pbmc_sce_percellqc <- scater::addPerFeatureQC(pbmc_sce_percellqc)
+
+
+# Filter out 0s, and see before/after dims:
+dim(pbmc_sce_percellqc) 
+pbmc_sce_percellqc <- pbmc_sce_percellqc[, pbmc_sce_percellqc$total > 0]
+dim(pbmc_sce_percellqc) 
+```
+
+
+Before proceeding with further filtering, we'll want to re-arrange the SCE object to handle the RNA and ADT separately.
+Currently, the ADTs and genes are both part of the primary experiment, so let's move the ADT counts into an altExp slot.
+
+```{r}
+gene_names <- as.data.frame(rowData(pbmc_sce_percellqc)) %>% 
+  dplyr::filter(Type == "Gene Expression") %>% 
+  dplyr::pull(ID)
+adt_names <- as.data.frame(rowData(pbmc_sce_percellqc)) %>% 
+  dplyr::filter(Type == "Antibody Capture") %>% 
+  dplyr::pull(ID)
+
+adt_pbmc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% adt_names]
+
+# split out the experiments:
+pbmc_sce_percellqc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% gene_names]
+altExp(pbmc_sce_percellqc, "CITEseq") <- adt_pbmc
+```
+
+
+
+### Filter RNA
+
+#### First, with `DropletUtils`
+
+```{r}
+# Filter RNA
+droplet_df <- DropletUtils::emptyDrops(
+  counts(pbmc_sce_percellqc), 
+  BPPARAM = BiocParallel::MulticoreParam(4)
+) 
+
+# Filter down to low FDR cells and check out dimensions
+# Values here are mostly NA.
+# NOTE: this filtering does propagate to altExp!! :) 
+cells_to_retain <- which(droplet_df$FDR <= 0.01)
+filtered_sce <- pbmc_sce_percellqc[, cells_to_retain]
+dim(filtered_sce) 
+```
+
+
+#### We can also filter RNA with `miQC`
+
+Mito QC stats are already present in `pbmc_sce_percellqc`.
+This is sometimes working and sometimes not for me, so commenting out for now...
+```{r}
+#miqc_model <- miQC::mixtureModel(pbmc_sce_percellqc) 
```

Comment:
We will want to do both; they serve different functions. First filter empty droplets with emptydops, then compromised (dead) cells with miQC. So miQC should use the filtered data as input.

Reply:
This fixes itself once the right input is provided!